tests/test_database.py

```diff
@@ -0,0 +1,105 @@
+import random
+import time
+from contextlib import contextmanager
+
+import pytest
+
+
+@pytest.fixture(scope='function')
+def db_sa(db):
+    # To make it easier to test, we keep the test restricted to firebase_tests
+    # Because of the current mutations on calls, we return it as a function.
+    name = 'test_%05d' % random.randint(0, 99999)
+    yield lambda: db().child(name)
+
+
+@contextmanager
+def make_stream(db, cbk):
+    s = db.stream(cbk)
+    try:
+        yield s
+    finally:
+        s.close()
+
+
+@contextmanager
+def make_append_stream(db):
+    l = []
+
+    def cbk(event):
+        l.append(event)
+
+    with make_stream(db, cbk) as s:
+        yield s, l
+
+
+class TestSimpleGetAndPut:
+    def test_simple_get(self, db_sa):
+        assert db_sa().get().val() is None
+
+    def test_put_succeed(self, db_sa):
+        assert db_sa().set(True)
+
+    def test_put_then_get_keeps_value(self, db_sa):
+        db_sa().set("some_value")
+        assert db_sa().get().val() == "some_value"
+
+    def test_put_dictionary(self, db_sa):
+        v = dict(a=1, b="2", c=dict(x=3.1, y="3.2"))
+        db_sa().set(v)
+
+        assert db_sa().get().val() == v
+
+    @pytest.mark.skip
```

Comment:
@thisbejim 
I get weird result on this test, do you see anything wrong with the code?